src/Plugin/Commerce/PaymentGateway/OffsiteRedirect.php

```diff
@@ -181,7 +182,9 @@ public function onNotify(Request $request):Response|null {
     }
 
     // 2. Check if the order can be loaded from the 'paymentRequestNumber'.
-    $order_id = $data['paymentRequestNumber'] ?? 0;
+    // Remove separator and suffix (if they exist) from paymentRequestNumber.
+    $order_id = explode(PaymentOffsiteForm::ORDER_ID_SUFFIX_SEPARATOR, $data['paymentRequestNumber'])[0]  ?? 0;
```

Comment:
This looks good, could you remove that extra space before the `??`? And I know you can do array access off the `explode()` function but could you instead separate those two so it's easier to read?

Reply:
Thank you very much for your comment. I have followed your advice to make changes and pushed.